unittests/test_utils.hpp

```diff
@@ -85,56 +118,56 @@ auto set_code(eosio::chain::controller& control, account_name account, const vec
                                .vmversion = 0,
                                .code = bytes(wasm.begin(), wasm.end())
                             });
-   return push_input_trx(control, account, trx);
+   return push_input_trx(app, control, account, trx);
 }
 
 void activate_protocol_features_set_bios_contract(appbase::scoped_app& app, chain_plugin* chain_plug) {
    using namespace appbase;
 
-   std::promise<void> feature_promise;
-   std::future<void> feature_future = feature_promise.get_future();
-   app->executor().post( priority::high, exec_queue::read_write, [&chain_plug=chain_plug, &feature_promise](){
-      const auto& pfm = chain_plug->chain().get_protocol_feature_manager();
-      auto preactivate_feature_digest = pfm.get_builtin_digest(builtin_protocol_feature_t::preactivate_feature);
-      BOOST_CHECK( preactivate_feature_digest );
-      chain_plug->chain().preactivate_feature( *preactivate_feature_digest, false );
-      std::vector<builtin_protocol_feature_t> pfs{
-         builtin_protocol_feature_t::only_link_to_existing_permission,
-         builtin_protocol_feature_t::replace_deferred,
-         builtin_protocol_feature_t::no_duplicate_deferred_id,
-         builtin_protocol_feature_t::fix_linkauth_restriction,
-         builtin_protocol_feature_t::disallow_empty_producer_schedule,
-         builtin_protocol_feature_t::restrict_action_to_self,
-         builtin_protocol_feature_t::only_bill_first_authorizer,
-         builtin_protocol_feature_t::forward_setcode,
-         builtin_protocol_feature_t::get_sender,
-         builtin_protocol_feature_t::ram_restrictions,
-         builtin_protocol_feature_t::webauthn_key,
-         builtin_protocol_feature_t::wtmsig_block_signatures };
-      for (const auto t : pfs) {
-         auto feature_digest = pfm.get_builtin_digest(t);
-         BOOST_CHECK( feature_digest );
-         chain_plug->chain().preactivate_feature( *feature_digest, false );
-      }
-      feature_promise.set_value();
-   });
+   std::atomic<bool> feature_set = false;
+   // has to execute when pending block is not null
```

Comment:
Same here, I think we should pass a `shared_ptr` to ` std::atomic<bool>` to make sure the lambda doesn't reference a non-existing stack frame.

Reply:
Done.